### ISSUE TITLE
"regr.xgboost" learner not run in parallel

### ISSUE THREAD

**Issue by caprone** [mlr-org/mlr#2470]
HI
"regr.xgboost" algorithm doesn't run in parallel:

```r
library(tidyverse)
library(mlr)
library(parallelMap)

tb = as_tibble(replicate(10, runif(100000)))

task = makeRegrTask(id = "toy", data = tb, target = "V10")
lrn = makeLearner("regr.xgboost") 
lrn = setHyperPars(lrn, nrounds=500, nthread=8)

# on Windows
parallelStartSocket(10)
mod = train(lrn, task)
```

**Comment by larskotthoff** [comment]
The parallelization that mlr provides does not allow you to parallelize learners that are not running in parallel already -- what it does is allow you to run multiple, independent evaluations (e.g. for benchmarking) in parallel.

If xgboost doesn't use multiple cores despite your settings, there may be something wrong with the xgboost setup. In any case, there's nothing mlr can do about it.

**Comment by caprone** [comment]
I'm sorry but I do not agree with You;
R standard xgboost implementation (xgboost package)  run in parallel, also with parallelMap package.

**Comment by berndbischl** [comment]
1)
well the thing is: mlr just directly calls into xgboost.
have you looked at the code here:
https://github.com/mlr-org/mlr/blob/master/R/RLearner_regr_xgboost.R

there really is nothing special mlr does (especially wrt to running xgb parallel)

2) 
you should not need the call to parallelStartSocket. if you leave that out, does that remove the problem?

@ja-thomas @mb706 do you have further infos here?

**Comment by pat-s** [comment]
From `xgboost::xgboost`:

> Parallelization is automatically enabled if OpenMP is present. Number of threads can also be manually specified via nthread parameter.

Do you have OpenMP support enabled?

Looking at it, I don't think that this is _mlr_ specific but only depends on your OpenMP setup since its "automatically" enabled. _mlr_ just let's you set the the number of threads used. 

Your code works for me (tested on Linux though):

![selection_004](https://user-images.githubusercontent.com/8430564/48076038-b3322080-e1e4-11e8-8001-67b8f36dd00f.png)

**Comment by caprone** [comment]
Yes, thanks! ;
Sorry again!!, probably  0.71.2 version  issue (on Win), last available on CRAN ;
I have solved  with  a downgrade to 0.6 version, until 0.81 version will be available from CRAN.

thanks again!

**Comment by pat-s** [comment]
> Sorry again!!, probably 0.71.2 version issue (on Win), last available on CRAN ;
> I have solved with a downgrade to 0.6 version, until 0.81 version will be available from CRAN.

You mean an R pkg (xgboost) here? OpenMP is a system library that needs to be enabled during the installation of R. Once it works it should be pkg version independent.

**Comment by caprone** [comment]
Yes, R package -- xgboost --
